### PR TITLE
update ci configuration to run on self-hosted runner

### PR DESCRIPTION
**Pull request by yaswant**
Update: CI build and test should now use Met Office self-hosted runner

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,54 +8,56 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: jedi-self-hosted-rg
```
Comment (mikecooke77) [inline .github/workflows/ci.yml:11]: This name seems a bit hard to interpret to me.  Why not:
jedi-self-hosted-runner

File: ci/build-and-test [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+THIS="$(basename "$0")"
+NPROC=${NPROC:-$(nproc)}
+WORKD="$(mktemp -d "${THIS}-XXXXXX" -t)"
+
+trap finally ERR
+trap finally EXIT
+
+cd "${WORKD}"
+
+# - jcsda container requires manual sourcing of spack environment
```
Comment (mikecooke77) [inline ci/build-and-test]: Do we need to keep these comment lines?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name: CI
+name: UKMO_CI_Az
```
Comment (mikecooke77) [inline .github/workflows/ci.yml]: Why not a little more explicit:
UKMO_CI_Azure

**Review by mikecooke77 — COMMENTED** [review]
Looks good just a couple of comments

**Review by matthewrmshin — APPROVED** [review]
LGTM.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (yaswant) [reply .github/workflows/ci.yml:11]: We have two action runner groups (rg): 
1. actions-rg (for MetOffice repos) 
2. actions-rg-ext (for externally hosted repos)

I've randomly chosen the names (jedi-self-hosted-rg, jedi-self-hosted-ext) over default action-runner during configuration time.  In order to change these, we need to go through the process again.. Hopefully we can sort this out in a future PR.

File: ci/build-and-test (hunk 2 above)
Comment (yaswant) [reply ci/build-and-test]: No. removed now.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (yaswant) [reply .github/workflows/ci.yml]: Done.

**Review by mikecooke77 — APPROVED** [review]

**Pull request merged**